assets/app/scripts/services/data.js

```diff
@@ -815,59 +809,69 @@ DataService.prototype.createStream = function(resource, name, context, opts, isR
     return true;
   }
 
-  DataService.prototype._isTooManyWebsocketRetries = function(resource, context) {
-    var key = this._uniqueKeyForResourceContext(resource, context);
+  DataService.prototype._isTooManyWebsocketRetries = function(key) {
     var events = this._websocketEventsMap[key];
     if (!events) {
       return false;
     }
 
     if (isTooManyRecentEvents(events)) {
-      Logger.log("Too many websocket open or close events for resource/context in a short period", resource, context, events);
+      Logger.log("Too many websocket open or close events for resource/context in a short period", key, events);
       return true;
     }
 
     if (isTooManyConsecutiveCloses(events)) {
-      Logger.log("Too many consecutive websocket close events for resource/context", resource, context, events);
+      Logger.log("Too many consecutive websocket close events for resource/context", key, events);
       return true;
     }
 
     return false;
   };
 
-  DataService.prototype._uniqueKeyForResourceContext = function(resource, context) {
-    // Note: when we start handling selecting multiple projects this
-    // will change to include all relevant scope
-    if (resource.equals("projects") || resource.equals("projectrequests")) { // when we are loading non-namespaced resources we don't need additional context
-      return resource;
-    }
-    else if (context.namespace) {
-      return resource.toString() + "/" + context.namespace;
-    }
-    else if (context.project && context.project.metadata) {
-      return resource.toString() + "/" + context.project.metadata.name;
-    }
-    else if (context.projectName) {
-      return resource.toString() + "/" + context.projectName;
-    }
-    else {
-      return resource.toString();
-    }
+
+  // will take an object, fitler & sort it for consistent unique key generation.
+  var paramsForKey = function(params) {
+    var sortedWhiteList = _.keysIn(
+                            _.pick(
+                              params, ['fieldSelector', 'labelSelector'])
+                          ).sort();
+    return _.object(
```

Comment:
there is no guarantee that an object has stable order, @liggitt found this for chrome https://bugs.chromium.org/p/v8/issues/detail?id=164

you'll need to approach stable param ordering another way, appending them to the key yourself by iterating over your sortedWhiteList array is one approach